.github/workflows/analyzers.yaml

```
@@ -0,0 +1,95 @@
+name: Run analyzers
+
+on:
+  push:
+    branches: [main]
+  pull_request:
+    branches: [main]
+  workflow_dispatch:
+
+jobs:
+  cppcheck:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Install dependencies
+        run: |
+          sudo apt-get update
+          sudo DEBIAN_FRONTEND=noninteractive apt-get -y install cppcheck
+
+      - name: Configure
+        run: |
+          cmake -S . -B build                     \
+            -DCMAKE_EXPORT_COMPILE_COMMANDS=ON    \
+            -DBUILD_PYTHON=OFF                    \
+            -DBUILD_TESTING=OFF
+
+      - name: Run cppcheck
+        run: |
+          cppcheck                                          \
+            --enable=style,portability,performance,warning  \
+            --library=posix                                 \
+            --library=cppcheck/segyio.cfg                   \
+            --suppressions-list=cppcheck/suppressions.txt   \
+            --inline-suppr                                  \
+            --project=build/compile_commands.json           \
+            --error-exitcode=1
+
+  scan_build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Install dependencies
+        run: |
+          sudo apt-get update
+          sudo DEBIAN_FRONTEND=noninteractive apt-get -y install clang clang-tools libfindbin-libs-perl
+
+      - name: Configure segyio
+        run: |
+          sudo scan-build --status-bugs cmake \
+            -S .                              \
+            -B build                          \
+            -DCMAKE_BUILD_TYPE=Release        \
+            -DBUILD_PYTHON=OFF                \
+            -DBUILD_TESTING=OFF
+
+      - name: Run scan-build
+        run: |
+          sudo scan-build --status-bugs cmake  \
+            --build build                      \
+            --target install
+
+  valgrind:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Install dependencies
+        run: |
+          sudo apt-get update
+          sudo DEBIAN_FRONTEND=noninteractive apt-get -y install valgrind
+
+      - name: Build segyio
+        run: |
+          cmake -S . -B build           \
+            -DCMAKE_BUILD_TYPE=Release  \
```

```suggestion
            -DCMAKE_BUILD_TYPE=Debug  \
```

I think running Valgrind on a debug build gives fewer false positives and better information (stack traces) if something goes wrong.

From [Valgrind's quickstart guide (point 2)](https://valgrind.org/docs/manual/quick-start.html):

> Compile your program with -g to include debugging information so that Memcheck's error messages include exact line numbers. Using -O0 is also a good idea, if you can tolerate the slowdown. With -O1 line numbers in error messages can be inaccurate, although generally speaking running Memcheck on code compiled at -O1 works fairly well, and the speed improvement compared to running -O0 is quite significant. Use of -O2 and above is not recommended as Memcheck occasionally reports uninitialised-value errors which don't really exist.